### PR TITLE
chore: update tidb-dashboard to v2023.09.21.1

### LINKED ISSUE #7137: dashboard: fix security issues
## Development Task

Related tidb-dashboard release: https://github.com/pingcap/tidb-dashboard/pull/1596

```
- security fixes
- hide full stacktrace info in error message
```

### PR DESCRIPTION
**Pull request by baurine**
### What problem does this PR solve?

Issue Number: Close #7137

### What is changed and how does it work?

Related tidb-dashboard release: https://github.com/pingcap/tidb-dashboard/pull/1596

```
- security fixes
- hide full stacktrace info in error message
```

### Release note

```release-note
None.
```

### Test

```
$ tiup playground v6.5.5 --tiflash 0 --pd.binpath /mnt/bao/codes/work/pd/bin/pd-server
```

image

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -28,7 +28,7 @@ require (
 	github.com/pingcap/kvproto v0.0.0-20230726063044-73d6d7f3756b
 	github.com/pingcap/log v1.1.1-0.20221110025148-ca232912c9f3
 	github.com/pingcap/sysutil v0.0.0-20211208032423-041a72e5860d
-	github.com/pingcap/tidb-dashboard v0.0.0-20230911054332-22add1e00511
+	github.com/pingcap/tidb-dashboard v0.0.0-20230921035205-31ea86f64e20
```
Comment (HuSharp) [inline go.mod:31]: I noticed it's `release` brach, do we need to base on `release-6.5` branch?

**Review by rleungx — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (baurine) [reply go.mod:31]: Nope, tidb-dashboard `release-6.5` branch is used for individual deployment, not for PD.

For PD, tidb-dashboard always use the internal version (version is the date format).

**Pull request merged**